### PR TITLE
using _ instead of . for app resource

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Description

We have been using the incorrect format for app resource since implementation. 
Updated all places where resource for an altinn app is defined to use `_` instead of `.`

## Related Issue(s)
- #495 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: .github/workflows/build-and-analyze.yml [hunk 1]
```diff
@@ -54,32 +36,34 @@ jobs:
...
+      - name: Complete sonar analysis
+        if: always()
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: |
+          dotnet-sonarscanner end /d:sonar.token="${{ secrets.SONAR_TOKEN }}"
+      - name: Process .NET test result
+        if: always()
+        uses: NasAmin/trx-parser@v0.5.0
```
Comment (SandGrainOne) [inline .github/workflows/build-and-analyze.yml:66]: A new third party action:
https://pedia.altinn.cloud/sec/activities/tools/github-actions/

Comment (acn-sbuad) [reply .github/workflows/build-and-analyze.yml:66]: This action is used across many of our repositories already, events was just lagging behind

File: src/Events/Authorization/SubscriptionXacmlMapper.cs [hunk 2]
```diff
@@ -72,7 +72,7 @@ private static XacmlJsonCategory CreateResourceCategory(Subscription subscriptio
 
             if (isAppEventSubs)
             {
-                string[] resourceAttParts = resourceFilterValue.Split('.');
+                string[] resourceAttParts = resourceFilterValue.Split('_');
```
Comment (SandGrainOne) [inline src/Events/Authorization/SubscriptionXacmlMapper.cs:75]: Could it be that apps are more likely to have '_' in their name? Is it possible to have '_' in other parts of the urn

Comment (SandGrainOne) [reply src/Events/Authorization/SubscriptionXacmlMapper.cs:75]: nvm, tried to create an app with _, not allowed

**Review by SandGrainOne — APPROVED** [review]

**Pull request merged**